### PR TITLE
B-19728-INT customer upload additional docs

### PR DESCRIPTION
**Pull request by paulstonebraker**
## Summary

Give the customers the ability to upload documents without affecting their workflow!

This work create a new button for customers that appears on their MoveHome page after a move is submitted, that allows them to provide additional documentation to the government without affecting their workflow. We added a new column to the moves table that houses additional_documents_id. Customer have the ability to upload documents and delete documents after they've been uploaded.

### How to test

1. Login as a customer with a move that has been submitted, and navigate to the MoveHome page of a move.
2. Click on the Upload Additional Documents button attached to the Profile step.
3. Confirm the page looks good and the upload/delete works, and persists as you navigate away/log out
4. Create new moves and add documents to new moves and previous moves, and make sure they're attaching to the correct move
5. Try and break it

### CODE REVIEW

**Review by loganwc — APPROVED** [review]
did what I could not 🙌

**Comment by pambecker** [comment]
Shouldn't there be a checkin for the FF in the milmove-feature-flags repository also?  or was that already done?

File: src/pages/MyMove/AdditionalDocuments/AdditionalDocuments.jsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            {warningMessage}
+          </Alert>
+        </Grid>
+      </Grid>
+      <Grid row data-testid="upload-info-container">
+        <Grid col desktop={{ col: 8, offset: 2 }}>
+          <SectionWrapper>
+            <h5>Upload documents</h5>
+            <Hint>PDF, JPG, or PNG only. Maximum file size 25MB. Each page must be clear and legible</Hint>
+            {/* {uploads?.length > 0 && ( */}
```
Comment (pambecker) [inline src/pages/MyMove/AdditionalDocuments/AdditionalDocuments.jsx:106]: can this be removed, and the one on line 111 and 122?

**Comment by paulstonebraker** [comment]
Moving back to draft to address bug

File: .envrc [hunk 2]
```diff
@@ -133,6 +133,9 @@ export FEATURE_FLAG_MOVE_LOCK=false
 export FEATURE_FLAG_OKTA_DODID_INPUT=false
 export FEATURE_FLAG_SAFETY_MOVE=true
 
+# Feature flag for additional supporting documents uploaded by customer
+export FEATURE_FLAG_MANAGE_SUPPORTING_DOCS=true
```
Comment (deandreJones) [inline .envrc:137]: I don't see this FF in the feature flag repo

Comment (paulstonebraker) [reply .envrc:137]: Thanks, Pam reminded me on Friday but I forgot amidst bug fix found during her review. Will push to ff repo shortly

File: pkg/handlers/internalapi/uploads.go [hunk 3]
```diff
@@ -162,6 +162,32 @@ func (h DeleteUploadHandler) Handle(params uploadop.DeleteUploadParams) middlewa
...
 			}
+
+			if params.MoveID != nil {
+				moveID, e := uuid.FromString(params.MoveID.String())
+				if e != nil {
+					appCtx.Logger().Error(fmt.Sprintf("UUID Parsing for %s", moveID.String()), zap.Error(err))
+					return handlers.ResponseForError(appCtx.Logger(), e), e
+				}
+
+				if e != nil {
```
Comment (pambecker) [inline pkg/handlers/internalapi/uploads.go:173]: this is duplicate from the if statement above, correct?

Comment (paulstonebraker) [reply pkg/handlers/internalapi/uploads.go]: Yep good eye! Removed

**Review by pambecker — APPROVED** [review]

**Pull request merged**